presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraRecordSink.java

```diff
@@ -81,16 +88,18 @@ public void beginRecord()
 
         field = 0;
         values.clear();
-        values.add(UUID.randomUUID());
+        if (generateUUID) {
+            values.add(UUID.randomUUID());
+        }
     }
 
     @Override
     public void finishRecord()
     {
         checkState(field != -1, "not in record");
-        checkState(field == fieldCount, "not all fields set");
+        checkState(field == columnTypes.size(), "not all fields set");
         field = -1;
-        cassandraSession.execute(insertQuery, values.toArray());
+        cassandraSession.executeWithSession(s -> s.execute(insert.bind(values.toArray())));
```

Comment:
Also, I don't see that Insert has a .bind() method in the java docs I'm looking at, but I trust that the code compiles; probably the more recent version of the driver includes that.